client-js/src/events.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  CLValueParsers,
+  CLMap,
+  CLString,
+  CLValueBuilder,
+  CLTypeTag,
+  CLValue,
+  CasperServiceByJsonRPC,
+} from "casper-js-sdk";
+
+import { Parser } from "@make-software/ces-js-parser";
+
+import { EventItem, EventParsed, CEP47Events, Transform } from "./types";
+
+export const CEP47EventParserFactory =
+  ({
+    contractPackageHash,
+    eventNames,
+  }: {
+    contractPackageHash: string;
+    eventNames: CEP47Events[];
+  }) =>
+  (value: EventItem) => {
+    if (value.body.DeployProcessed.execution_result.Success) {
+      const { transforms } =
+        value.body.DeployProcessed.execution_result.Success.effect;
+
+      const cep47Events = transforms.reduce(
+        (acc: EventParsed[], val: Transform) => {
+          if (val.transform.WriteCLValue?.cl_type === "Any") {
+            const maybeCLValue = CLValueParsers.fromBytesWithType(
+              Buffer.from(val.transform.WriteCLValue?.bytes, "hex")
+            );
+            const clValue = maybeCLValue.unwrap();
+
+            if (clValue?.clType().tag === CLTypeTag.Map) {
+              const hash = (clValue as CLMap<CLValue, CLValue>).get(
+                CLValueBuilder.string("cep78_contract_package")
```

Comment:
let's please extract this string to a const ?

Reply:
I would go further and extract it to env variable ;)